web-client/src/app/views/two-factor-authentication/two-factor-authentication.page.ts

```diff
@@ -56,7 +63,11 @@ export class TwoFactorAuthenticationPage implements OnInit {
     });
   }
 
-  ngOnInit() {}
+  ngOnInit() {
+    this.sessionQuery.wallet.pipe(untilDestroyed(this)).subscribe((wallet) => {
```

Comment:
why don't you just subscribe to this on the constructor and pull the wallet id from it instead of subscribing on both `wallet` & `walletId`?